src/Spectre.AngularClient/src/app/preparations/preparation/preparation.component.ts

```diff
@@ -77,13 +83,31 @@ export class PreparationComponent implements OnInit {
       .get(this.id, this.currentChannelId)
       .subscribe(heatmap => this.heatmapData = this.toHeatmapDataset(heatmap));
   }
+
+  onChangedXCoordinate(event: any) {
+    this.xCoordinate = event.value;
+    this.getSpectrumByCoordianates();
+  }
+
+  onChangedYCoordinate(event: any) {
+    this.yCoordinate = event.value;
+    this.getSpectrumByCoordianates();
+  }
+
   getSpectrum(selectNumber: number) {
     this.spectrumService
       .get(this.id, selectNumber)
       .subscribe(spectrum => this.spectrumData = this.toSpectrumDataset(spectrum));
   }
 
+  getSpectrumByCoordianates() {
+    this.spectrumService
+      .getByCoordinates(this.id, this.xCoordinate + this.minHeatmapColumn, (this.yCoordinate - this.yHeatmapSize)*(-1) + this.minHeatmapRow)
+      .subscribe(spectrum => this.spectrumData = this.toSpectrumDataset(spectrum), error => this.showError("Spectrum not found"));
```

Comment:
The `error` path is just added as a string somewhere in the page. I think we should either go for popup, or e.g. change colors and disable spectrum plot.